packages/flutter/lib/src/rendering/sliver_fixed_extent_list.dart

```diff
@@ -170,28 +200,60 @@ abstract class RenderSliverFixedExtentBoxAdaptor extends RenderSliverMultiBoxAda
     return trailingGarbage;
   }
 
+  int _getChildIndexForScrollOffset(double scrollOffset, ItemExtentGetter callback) {
+    if (scrollOffset == 0.0) {
+      return 0;
+    }
+    double position = 0.0;
+    int index = 0;
+    while (true) {
+      if (position >= scrollOffset) {
+        break;
+      }
+      position += callback(index, _currentLayoutDimensions);
+      ++index;
+    }
+    return index - 1;
+  }
+
+  BoxConstraints _getChildConstraints(int index) {
+    double extent;
+    if (itemExtentCallback == null) {
+      extent = itemExtent;
+    } else {
+      extent = itemExtentCallback!(index, _currentLayoutDimensions);
+    }
+    return constraints.asBoxConstraints(
+      minExtent: extent,
+      maxExtent: extent,
+    );
+  }
+
+  late SliverLayoutDimensions _currentLayoutDimensions;
+
   @override
   void performLayout() {
     final SliverConstraints constraints = this.constraints;
     childManager.didStartLayout();
     childManager.setDidUnderflow(false);
 
-    final double itemExtent = this.itemExtent;
-
+    final double itemFixedExtent = itemExtent;
     final double scrollOffset = constraints.scrollOffset + constraints.cacheOrigin;
     assert(scrollOffset >= 0.0);
     final double remainingExtent = constraints.remainingCacheExtent;
     assert(remainingExtent >= 0.0);
     final double targetEndScrollOffset = scrollOffset + remainingExtent;
 
-    final BoxConstraints childConstraints = constraints.asBoxConstraints(
-      minExtent: itemExtent,
-      maxExtent: itemExtent,
+    _currentLayoutDimensions = SliverLayoutDimensions(
+        scrollOffset: constraints.scrollOffset,
+        precedingScrollExtent: constraints.precedingScrollExtent,
```

Comment:
Ah ok, it looks like this is not updated as each child is laid out. I think users will expect it to account for the children in the SliverList that came before it in addition to any slivers that came before it. Maybe SliverLayoutDimensions can have a copyWith method, and we can add each child's extent to preceding extent as we lay them out and call on the item extent getter?

Reply:
>I think users will expect it to account for the children in the SliverList that came before it in addition to any slivers that came before it.
If I understand correctly, you mean when call `itemExtentBuilder(1000, dimentions)`, `dimentions.precedingScrollExtent` should equal `constraints.precedingScrollExtent + sumExtentsOf(0~999)`?

If that's the case, developers can calculate this value themselves, right? Or the `dimentions.scrollOffset` is more meaningful? Otherwise introducing a new `precedingScrollExtent` would be confused with `SliverConstraints.precedingScrollExtent`. 

Additional, during a frame, `itemExtentBuilder` will be called many times, and the called item does not mean that it will be loaded/layout by the lazy loading mechanism. I mean, when scrolling to the end, all items‘s callback will be called.

Reply:
Oh you are right, and make a really good point about opportunities for confusion there. The user could also keep track of the extents as the builder is called. Thanks for explaining!

Reply:
Also! I just thought, maybe it should be documented somewhere that the itemExtentBuilder could be called multiple times so users are aware 👍

Reply:
Added some docs to `flutter.widgets.list_view.itemExtentBuilder`

```dart
  ///
  /// This will be called multiple times during the layout phase of a frame to find
  /// the items that should be loaded by the lazy loading process.
  ///

```